### PR TITLE
more adjustments to PR template

### PR DESCRIPTION
**Pull request by clintropolis**
Some additional adjustments after a bit more time using the template, with the main change being to put all continuous text blocks on one line so they flow better and are easier to select and remove in the description editor of the github UI.

Anyone have any thoughts on also adding some words of encouragement near the beginning of the template to help remind people why they are being asked to do all this? Maybe something like:

_Thanks for trying to help us make Apache Druid (incubating) be the best it can be! Please fill out as much of the following information as is possible (where relevant) to help make the intention and scope of this PR clear in order to ease review._

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -51,12 +42,10 @@ This PR has:
 - [ ] added integration tests.
```
Comment (ccaominh) [inline .github/pull_request_template.md]: Since you're updating the template, line 37 (2nd item in PR checklist) has an extra indent that can be removed

Comment (clintropolis) [reply .github/pull_request_template.md]: If you're referring to the 'concurrency checklist' thing, i believe it's supposed to be a sub-item of 'self reviewed'

Comment (clintropolis) [reply .github/pull_request_template.md]: That said, I'm open to changing it, either way seems fine to me

Comment (ccaominh) [reply .github/pull_request_template.md]: Ah, if it's a sub item of "self reviewed", then it's fine as is.

**Comment by gianm** [comment]
> Thanks for trying to help us make Apache Druid (incubating) be the best it can be! Please fill out as much of the following information as is possible (where relevant) to help make the intention and scope of this PR clear in order to ease review.

Something like this would be nice, how about also mentioning that in addition filling in relevant things, people should delete things that are _irrelevant_.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,41 +1,32 @@
 Fixes #XXXX.
 
-(Replace XXXX with the id of the issue fixed in this PR. Remove the above line if there is no corresponding
-issue. Don't reference the issue in the title of this pull-request.)
+_Replace XXXX with the id of the issue fixed in this PR. Remove this section if there is no corresponding issue. Don't reference the issue in the title of this pull-request._
```
Comment (gianm) [inline .github/pull_request_template.md]: Since this is going to show up in the edit view, parens probably make more sense than underscores.

**Pull request merged**